commcare_connect/opportunity/forms.py

```diff
@@ -1158,45 +1163,80 @@ def __init__(self, *args, **kwargs):
+                            "amount",
                             label=f"Amount ({self.opportunity.currency})",
                             **{
+                                "x-model": "amount",
                                 "x-ref": "amount",
                                 "x-on:input.debounce.300ms": "convert()",
+                                ":readonly": "serviceDeliverySelected()",
+                            },
+                        ),
+                        Field(
+                            "amount_usd",
+                            **{
+                                "x-model": "usdAmount",
+                                ":readonly": "serviceDeliverySelected()",
                             },
                         ),
-                        Field("amount_usd"),
                         css_class="grid grid-cols-2 gap-4",
                     ),
                     Div(css_id="converted-amount-wrapper", css_class="space-y-1 text-sm text-gray-500 mb-4"),
                     css_class="flex flex-col",
                 ),
                 Div(
-                    Field(
-                        "start_date",
+                    Div(
+                        Field(
+                            "start_date",
+                            **{
+                                "x-on:change": "fetchInvoiceLineItems()",
+                                "x-model": "startDate",
+                            },
+                        ),
+                        Field(
+                            "end_date",
+                            **{"x-on:change": "fetchInvoiceLineItems()", "x-model": "endDate"},
+                        ),
+                        css_class="grid grid-cols-2 gap-4",
                     ),
-                    Field(
-                        "end_date",
+                    Fieldset(
+                        "Line Items",
+                        Div(
+                            css_id="invoice-line-items-wrapper",
+                            css_class="space-y-1 text-sm text-gray-500 mb-4",
+                        ),
                     ),
-                    **{"x-show": "serviceDeliverySelected()"},
-                    css_class="grid grid-cols-2 gap-4",
-                ),
-                Div(
-                    Field("notes"),
+                    Fieldset(
+                        "Service Delivery Details",
+                        Field("notes"),
+                    ),
+                    css_class="flex flex-col",
                     **{"x-show": "serviceDeliverySelected()"},
                 ),
                 css_class="flex flex-col",
             ),
         )
         self.helper.form_tag = False
 
+    def get_earliest_uninvoiced_date(self):
+        date = (
+            CompletedWork.objects.filter(invoice__isnull=True, opportunity_access__opportunity=self.opportunity)
+            .aggregate(earliest_date=Min("date_created"))
```

Comment:
the dates we want to use for invoicing are the approval dates, which is `status_modified_date`